test/remove.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+
+
+test('remove', function() {
+
+  eq(typeof S.remove, 'function');
+  eq(S.remove.length, 2);
+  eq(S.remove.toString(), 'remove :: String -> StrMap a -> StrMap a');
+
+  function testBasic(key, strMap, strMapResult) {
+    eq(S.remove(key, strMap), strMapResult);
+  }
+
+  function testDoubleRemove(key, strMap) {
+    var remove = S.remove(key);
+    eq(remove(remove(strMap)), remove(strMap));
+  }
+
+  function inherit(parent, child) {
+    var inherited = Object.create(parent);
+    for (var property in child) {
+      inherited[property] = child[property];
+    }
+    return inherited;
+  }
+
+  function testRemove(key, strMap, strMapResult) {
+    testBasic(key, strMap, strMapResult);
+    testDoubleRemove(key, strMap);
+  }
+
+  testRemove('a', {}, {});
+  testRemove('a', {a: 1}, {});
+  testRemove('a', {b: 2}, {b: 2});
+  testRemove('b', {a: 1, b: 2}, {a: 1});
+  testRemove('a', {b: 2, c: 3}, {b: 2, c: 3});
+  testRemove('b', {a: 1, b: 2, c: 3}, {a: 1, c: 3});
+  // test inherited properties
+  testRemove('a', inherit({a: 1}, {}), {});
+  testRemove('b', inherit({a: 1}, {}), {a: 1});
+  testRemove('a', inherit({a: 1}, {a: 2}), {});
+  testRemove('b', inherit({a: 1}, {a: 2}), {a: 2});
+  testRemove('a', inherit({b: 2}, {a: 1}), {b: 2});
+  testRemove('a', inherit({a: 1}, {b: 2}), {b: 2});
+  testRemove('c', inherit({a: 1}, {b: 2}), {a: 1, b: 2});
+  testRemove('a', inherit({b: 2, c: 3}, {a: 1}), {b: 2, c: 3});
+  testRemove('b', inherit({b: 2, c: 3}, {a: 1}), {a: 1, c: 3});
```

Comment:
I don't see value in testing so many combinations. Would something like this suffice?

```javascript
function Point() {}
Point.prototype.x = 0;
Point.prototype.y = 0;

eq(S.remove('x', new Point()), {y: 0});
eq(S.remove('y', new Point()), {x: 0});
eq(S.remove('z', new Point()), {x: 0, y: 0});
```